rust/src/domains/polars/expr/mod.rs

```diff
@@ -0,0 +1,214 @@
+use polars::lazy::dsl::Expr;
+use polars::prelude::*;
+use std::collections::BTreeSet;
+use std::fmt::{Debug, Formatter};
+
+use crate::core::{Metric, MetricSpace};
+use crate::metrics::{
+    AbsoluteDistance, ChangeOneDistance, HammingDistance, InsertDeleteDistance, LInfDistance,
+    LpDistance, PartitionDistance, SymmetricDistance,
+};
+use crate::traits::ProductOrd;
+use crate::transformations::DatasetMetric;
+use crate::{core::Domain, error::Fallible};
+
+use super::{Frame, FrameDomain, LogicalPlanDomain, NumericDataType, SeriesDomain};
+
+#[cfg(feature = "ffi")]
+mod ffi;
+
+// TODO: remove this allow marker later
+#[allow(dead_code)]
+#[derive(Clone, PartialEq, Debug)]
+pub enum ExprContext {
+    // select
+    Select,
+    // filter
+    Filter,
+    // with_column
+    WithColumns,
+    // group_by
+    Aggregate(BTreeSet<String>),
+}
+
+impl ExprContext {
+    fn get_plan(&self, val: &(LogicalPlan, Expr)) -> LogicalPlan {
+        let (lp, expr) = val.clone();
+        let frame = LazyFrame::from(lp);
+        match self {
+            ExprContext::Select => frame.select([expr]),
+            ExprContext::Filter => frame.filter(expr),
+            ExprContext::WithColumns => frame.with_columns([expr]),
+            ExprContext::Aggregate(columns) => frame
+                .group_by(&columns.iter().map(|n| col(n)).collect::<Vec<_>>())
+                .agg([expr]),
+        }
+        .logical_plan
+    }
+
+    pub fn break_alignment(&self) -> Fallible<()> {
```

Comment:
I think I've seen this referenced in other PRs, but if we want each PR to stand on its own, then should there be some reference to this method here?